src/DocAssembler/README.md

```diff
@@ -0,0 +1,157 @@
+| `content` (Required)  | A collection of [`Content`](#content) objects to define all content sets to assemble. |
+
+### `Replacement`
+
+A replacement definition has these properties:
+
+| Property     | Description                                                  |
+| ------------ | ------------------------------------------------------------ |
+| `expression` | A regular expression to find specific text. |
+| `value` | The value that replaces the found text. Named matched subexpressions can be used here as well as explained below. |
+
+This type is used in collections for URL replacements or content replacements. They are applied one after another, starting with the first entry. The regular expression is used to find text that will be replaced by the value. The expressions are regular expression as described in [.NET Regular Expressions - .NET  Microsoft Learn](https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expressions). Examples can be found there as well. There are websites like [regex101: build, test, and debug regex](https://regex101.com/) to build, debug and validate the expression you need.
+
+#### Using named matched subexpressions
+
+Sometimes you want to find specific content, but also reuse parts of it in the value replacement. An example would be to find all `AB#1234` notations and replace it by a URL to the referenced Azure Boards work-item or GitHub item. But in this case we want to use the ID (1234) in the value. To do that, you can use [Named matched subexpressions](https://learn.microsoft.com/en-us/dotnet/standard/base-types/grouping-constructs-in-regular-expressions#named-matched-subexpressions).
+
+This expression could be used to find all those references:
+
+```regex
+(?<pre>[$\\s])AB#(?<id>[0-9]{3,6})
+```
+
+As we don't want to find a link like `[AB#1234](https://...)`, we look for all AB# references that are at the start of a line (using the `$` tag) or are prefixed by a whitespace (using the `\s` tag). As we need to keep that prefix in place, we capture it as a named subexpression called `pre`.
+
+> [!NOTE]
+>
+> As the expression is configured in a string in a JSON file, special characters like back-slashes need to be escaped by an (extra) back-slash.
+
+The second part is to get the numbers after the AB# text. This is configured here to be between 3 and 6 characters. We also want to reuse this ID in the value, so we capture it as a named subexpression called `id`.
+
+In the value we can reuse these named subexpression like this:
+
+```text
+${pre}[AB#${id}](https://dev.azure.com/[your organization]/_workitems/edit/${id})
+```
+
+We start with the `pre` value, after which we build a markdown link with AB# combined with the `id` as the text and the `id` as parameter for the URL. We reference an Azure Board work item here. Of course you need to replace the `[your organization]` with the proper value for your ADO environment here.
+
```

Comment:
so, maybe put the resulting json as an example

Reply:
Added.